ls1intum/Artemis
# `Exam mode`: Fix a bug that prevented the exam detail page from displaying anything after a reset

### Checklist
#### General


#### Client

### Motivation and Context



If an instructor resets an exam, the exam details disappear and they have to reload the page to continue working:

![grafik](https://user-images.githubusercontent.com/23171488/181792951-1801d002-6c19-457f-a6f1-e731ee33925c.png)


### Description


- Set course access rights on the exam returned by the server after reset so the authorization check succeeds and details are shown again
- Clearing an internal array of the exercise group list on change so it's freshly filled, otherwise the exercise groups of the reset exam would be appended
- Added an success alert for a successful reset as it's not super clear without it

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam with a few registered students

1. Generate student exams for the students
2. Prepare exercise start
3. Go to exam management view of the exam (with the checklist)
4. Verify that all exercise group in the exercise group list are listed correctly and have a number of participations
![grafik](https://user-images.githubusercontent.com/23171488/181793475-c7cbb946-f445-4fb0-bc39-29b976545e32.png)
5. Reset the exam
6. Verify that you get a success alert after you reset it and the modal closed
7. Verify that you can still see everything without reloading on the checklist and **don't** get an empty page like in the first image of this PR
8. Verify that the exercise group table does not contain duplicated exercises, and that the participants column shows nothing / only "-"

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-management.service.ts | 73% | 95% |
| exam-detail.component.ts | 75% | 96% |
| exam-checklist-exercisegroup-table.component.ts | 76% | 94% |
